app.py

```diff
@@ -107,9 +108,14 @@ def unauthorized():
     """Redirect unauthorized users to Login page."""
     flash('You must be logged in to view that page.')
     return redirect(url_for('login'))
+
 @app.route("/listings")
 def listings():
-    return render_template('listings.html', items=items)
+    prodTuple = db.getRetailers_Products()
+    products = []
+    for x in prodTuple:
+        products.append(ScrapedProduct(name = x[2], source = x[5], price = x[3], photo = x[4], instock= x[6], new = x[7], price_check= x[8]))
```

Comment:
I get what's going on here, but I think it might be better to have an object (we can even put it on this app.py document) that connects to the database and then just make one function call to the object from in here.  So something like:

```
class ProductDBInterface:
    def parse_sql_tuple(self, x):
        return (ScrapedProduct(name = x[2], source = x[5], price = x[3], photo = x[4], instock= x[6], new = x[7], price_check= x[8])

    def read_products_from_db(self):
        prod_tuple = db.getRetailers_Products()
        products = [self.parse_sql_into_scraped_product(x) for x in prod_tuple]
        return products

interface = ProductDBInterface()

```
....a few lines of code later...

```
@app.route("/listings")
def listings():
    return render_template('listings.html', items=interface.read_products_from_db())
    
```

Reply:
Just updated it. This is a lot prettier. Thanks :)